ctapipe/image/muon/muon_processor.py

```diff
@@ -0,0 +1,275 @@
+"""
+High level image processing  (ImageProcessor Component)
+"""
+import numpy as np
+from astropy.coordinates import SkyCoord
+from ctapipe.containers import TelEventIndexContainer
+
+from ctapipe.calib import CameraCalibrator
+from ctapipe.core import TelescopeComponent
+from ctapipe.core import traits
+from ctapipe.io import EventSource
+from ctapipe.image.cleaning import TailcutsImageCleaner
+from ctapipe.coordinates import TelescopeFrame, CameraFrame
+from ctapipe.containers import MuonParametersContainer, ArrayEventContainer
+from ctapipe.instrument import CameraGeometry, SubarrayDescription
+from ctapipe.image.muon import (
+    MuonRingFitter,
+    MuonIntensityFitter,
+    ring_containment,
+    ring_completeness,
+    intensity_ratio_inside_ring,
+    mean_squared_error,
+)
+
+
+class MuonProcessor(TelescopeComponent):
+    """
+    Takes Muon data and cleans and parametrizes the images.
+    """
+
+    completeness_threshold = traits.FloatTelescopeParameter(
+        default_value=30.0, help="Threshold for calculating the ``ring_completeness``"
+    ).tag(config=True)
+
+    ratio_width = traits.FloatTelescopeParameter(
+        default_value=1.5,
+        help=(
+            "Ring width for intensity ratio"
+            " computation as multiple of pixel diameter"
+        ),
+    ).tag(config=True)
+
+    overwrite = traits.Bool(
+        default_value=False, help="If true, overwrite outputfile without asking"
+    ).tag(config=True)
+
+    min_pixels = traits.IntTelescopeParameter(
+        help=(
+            "Minimum number of pixels after cleaning and ring finding"
+            "required to process an event"
+        ),
+        default_value=100,
+    ).tag(config=True)
+
+    pedestal = traits.FloatTelescopeParameter(
+        help="Pedestal noise rms", default_value=1.1
+    ).tag(config=True)
```

Comment:
Is this really needed as a parameter? Shouldn't it come from the monitoring info, as it changes over time during an observation and is pixel-dependent.

Reply:
This was taken from the earlier iteration of the muon analysis, but changing it to a pixel and time dependent pedestal should be the correct thing to do, at least on the long run (I don't know if the current muon simulations provide this information). How is this information stored?

Reply:
It should in principle be in `event.monitoring.tel[tel_id].pedestal`. But I don't think that we fill that with sensical values right now.

I also a bit confused of the different "pedestals" here. The monitoring pedestal should contain the average NSB in each pixel, right? And the `calibration.tel[dl1].pedestal_offset` is what is more commonly known as "baseline"?

Reply:
Looking at the containers again, this should be `event.monitoring.tel[tel_id].pedestal.charge_std`, which is currently not filled by the simtel event source, since it depends on the extractor used.

This is connected to the whole calibration chain and how to perform these calibration measurements on simulations, so I think it is out of scope for this PR and we should keep this as a parameter for now until that is worked out.